isy-serviceapi-core/src/main/java/de/bund/bva/isyfact/serviceapi/core/httpinvoker/TimeoutWiederholungHttpInvokerRequestExecutor.java

```diff
@@ -123,21 +126,27 @@ protected RemoteInvocationResult doExecuteRequest(HttpInvokerClientConfiguration
     @Override
     protected void prepareConnection(HttpURLConnection con, int contentLength) throws IOException {
         super.prepareConnection(con, contentLength);
-
-        if (aufrufKontextVerwalterOptional.isPresent()) {
-            String bearerToken = aufrufKontextVerwalterOptional.get().getBearerToken();
+        String bearerToken = null;
+        try {
+            if (aufrufKontextVerwalterOptional.isPresent()) {
+                bearerToken = aufrufKontextVerwalterOptional.get().getBearerToken();
+            } else if (SecurityContextHolder.getContext() != null && SecurityContextHolder.getContext().getAuthentication() != null) {
+                Authentication currentAuthentication = SecurityContextHolder.getContext().getAuthentication();
+                bearerToken = ((AbstractOAuth2TokenAuthenticationToken<?>) currentAuthentication).getToken().getTokenValue();
+            }
```

Comment:
Does it make sense to exchange the order on getting the baererToken?
At first using the new implementation of `SecurityContextHolder` even if the old implementation `AufrufKontextVerwalter` is present.
Only in case of no `SecurityContextHolder` the old implementation `AufrufKontextVerwalter` will be used.

Reply:
The process for obtaining the bearerToken has been modified. Initially, the new implementation of _SecurityContextHolder_ is examined, and if it is not set, the BearerToken from the AufurufKontextverwalter is utilized instead.

Reply:
solved